src/handlers/interventions-processor-handler.ts

```diff
@@ -40,141 +47,99 @@ export const handler = async (event: SQSEvent, context: Context): Promise<SQSBat
   }
 
   const itemFailures: SQSBatchItemFailure[] = [];
-  for (const record of event.Records) {
-    await processSQSRecord(itemFailures, record);
-  }
+  const promiseArray = event.Records.map((record: SQSRecord) => {
+    return processSQSRecord(record).catch(async (error) => {
+      const itemIdentifier = await handleError(error, record);
+      if (itemIdentifier) itemFailures.push({ itemIdentifier });
+    });
+  });
+  await Promise.allSettled(promiseArray);
```

Comment:
Those that reject would be because they threw an error, which has already been caught in the catch block in the loop. If we add the rejected promises at this point we would be double adding items